addons/mail/static/src/new/core/chat_window_model.js

```diff
@@ -32,8 +35,25 @@ export class ChatWindow {
             _state: state,
         });
         this.update(data);
-        state.chatWindows.push(this);
-        return state.chatWindows.find((c) => c.threadLocalId === data.threadLocalId); // return reactive version
+        let idx;
+        if (!data.replaceNewMessageChatWindow) {
+            idx = state.chatWindows.length;
+        } else {
+            const newMessageChatWindowIndex = state.chatWindows.findIndex(
+                ({ threadLocalId }) => !threadLocalId
+            );
+            idx = newMessageChatWindowIndex !== -1 ? newMessageChatWindowIndex : state.chatWindows.length;
+        }
+        state.chatWindows.splice(idx, 1, this);
+        return state.chatWindows[idx]; // return reactive version
+    }
+
+    get thread() {
+        return this._state.threads[this.threadLocalId];
+    }
+
+    get displayName() {
+        return this.thread?.displayName || _t("New message");
```

Comment:
Usually `??` is more appropriate than `||` for defaulting. This is especially true when `true`/`false` are valid values.